thrifty-schema/src/main/java/com/microsoft/thrifty/schema/Constant.java

```diff
@@ -258,6 +258,12 @@ public void validate(Linker linker, ThriftType expected, ConstValueElement value
                 } else {
                     throw new IllegalStateException("bad enum literal: " + value.value());
                 }
+            } else if (named instanceof Typedef) {
```

Comment:
If we do the typedef check this way, it looks like we'll miss validating the value of the enum.  How will we validate that the aliased type is the correct enum type?

I think a better approach would be to ensure that we don't look up a typedef in the first place.